invenio_rdm_records/ext.py

```diff
@@ -19,6 +22,25 @@
 from .services.schemas.metadata_extensions import MetadataExtensions
 
 
+def verify_token():
```

Comment:
true, the permission generator queries the links in `record.access.links` and only uses those that have the right permission level and haven't expired
so that case should be filtered